runtime/lua/vim/lsp/util.lua

```diff
@@ -1272,6 +1272,23 @@ function M.character_offset(buf, row, col)
   return str_utfindex(line, col)
 end
 
+function M.format_formatexpr()
+ -- only reformat on explicit gq command
+ -- `formatexpr` is also called when exceding
+ -- `textwidth` in insert mode
+ if vim.fn.mode() == 'i' or vim.fn.mode() == 'R' then
+    -- fall back to Vims internal reformatting
+    return 1
+ end
+ local start_line = vim.v.lnum
```

Comment:
I like to add these as arguments.

Then you can check if they're nil and create them the way you want like here. This way you can call the function on a range very easily:

`lsp.formatexpr(1, 25)` etc.